src/protocol.rs

```diff
@@ -310,26 +318,29 @@ impl ParticipantConfig {
     /// [`ParticipantIdentifier`]s.
     ///
     /// This method implies the existence of a trusted third party that
-    /// generates the IDs and gives them to participants.
+    /// generates the IDs; that's why it's only available.
     pub fn random_quorum<R: RngCore + CryptoRng>(
         size: usize,
         rng: &mut R,
-    ) -> Vec<ParticipantConfig> {
+    ) -> Result<Vec<ParticipantConfig>> {
+        if size < 2 {
+            Err(CallerError::ParticipantConfigError)?
+        }
         let ids = std::iter::repeat_with(|| ParticipantIdentifier::random(rng))
             .take(size)
             .collect::<Vec<_>>();
 
-        (0..size)
+        Ok((0..size)
             .map(|i| {
                 let mut other_ids = ids.clone();
                 let id = other_ids.swap_remove(i);
                 Self { id, other_ids }
             })
-            .collect()
+            .collect())
     }
 
-    #[cfg(test)]
-    fn random<R: RngCore + CryptoRng>(size: usize, rng: &mut R) -> ParticipantConfig {
+    /// Creating a random Participant Config
+    pub fn random<R: RngCore + CryptoRng>(size: usize, rng: &mut R) -> ParticipantConfig {
```

Comment:
nit: This can remain a private function configured for tests only. 

nit on docs: Thanks for adding docs here. Some general guidelines for writing docs is that they should be full sentences (e.g. ending with punctuation), and any Rust type we refer to should be in \`backticks\` and linked. So this would be
```rust
/// Create a random [`ParticipantConfig`].
```

Reply:
if I leave random a private function then it does not compile saying that random is never used.

Reply:
We discussed on a call, recording the outcome here for posterity:

There are 3 ways to address "never used" errors:
1. If the function is used in a test but not in the library, flag it with `#[cfg(test)]`
3. If the function is going to be used inside the library but you're not ready to do so yet, like if you're making a new internal module, flag it with `#[allow(unused)]`
2. If the function is supposed to be used by callers of the library and isn't used internally, make it public.

Otherwise, delete it :)